e2e/testmain_test.go

```diff
@@ -74,10 +80,25 @@ func TestMain(m *testing.M) {
 	parseFlagsOrDie()
 
 	ctx := context.Background()
-	client, err := google.DefaultClient(ctx, compute.ComputeScope)
+
+	credentials, err := google.FindDefaultCredentials(ctx, compute.ComputeScope)
 	if err != nil {
 		log.Fatal(err)
 	}
+	ts := credentials.TokenSource
+
+	if testFlags.serviceAccountName != "" {
+		ts, err = impersonate.CredentialsTokenSource(ctx, impersonate.CredentialsConfig{
+			TargetPrincipal: testFlags.serviceAccountName,
+			Scopes:          []string{compute.ComputeScope, compute.CloudPlatformScope},
+			Lifetime:        1200 * time.Second,
+			Delegates:       []string{},
+		}, option.WithCredentials(credentials))
+		if err != nil {
+			log.Fatal(err)
```

Comment:
please extend the log with more info + testFlags.serviceAccountName

Reply:
Done (although I believe that error messages in all functions called here are already sufficiently clear)